tf2/src/buffer_core.cpp

```diff
@@ -1455,12 +1449,8 @@ void BufferCore::testTransformableRequests()
         M_TransformableCallback::iterator it = transformable_callbacks_.find(req.cb_handle);
         if (it != transformable_callbacks_.end())
         {
-          transformables.push_back(boost::make_tuple(boost::ref(it->second),
```

Comment:
The objects referenced here might get destroyed before they are used below, resulting in a segfault.

Reply:
I need to look a little deeper that this isn't going to restore the potential deadlock of #91

Reply:
Sure. Fortunately, unit tests nicely cover this deadlock. While https://github.com/ros/geometry2/pull/539/commits/23e6295ebb14279892fd55b7141d6f50b7ed0f25 restores the deadlock (as expected), resulting in a `RLTestTimeoutException`, https://github.com/ros/geometry2/pull/539/commits/9fdf5474907bf1b9d0983ebdeb8b25c93a487960 fixes it again using the method described in #538.